jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/azurerules/DiskRule/CheckDiskEncryption.java

```diff
@@ -120,17 +120,15 @@ private boolean validateDiskEncryption(String esUrl, Map<String, Object> mustFil
                         .get("diskInner").getAsJsonObject();
                 if (diskInner != null) {
                     if (!isAttachedToVirtualMachine) {
-                        JsonObject encryptionSettings = diskInner.getAsJsonObject()
-                                .get("properties.encryptionSettings").getAsJsonObject();
-
-                        if (encryptionSettings == null) {
+                        if (diskInner.getAsJsonObject().get("properties.encryptionSettingsCollection").isJsonNull()) {
                             validationResult = false;
                         } else {
-                            boolean isEnabled = jsonDataItem.getAsJsonObject()
+                            JsonObject encryptionSettings = diskInner.getAsJsonObject()
+                                    .get("properties.encryptionSettingsCollection").getAsJsonObject();
```

Comment:
why don't you move these two lines outside of `if...else` end then either check if for null in `if` or convert it into json object in `else`?

Reply:
It was that way initially, just don't call `getAsJsonObject` above `if`

Reply:
As mentioned in the ticket (https://paladincloud.atlassian.net/browse/CORE-1613), the reason for the PR was null pointer exception at line 124, these two lines are in else block because we are not using these variables anywhere else apart from else block